cns/restserver/restserver.go

```diff
@@ -1252,7 +1263,31 @@ func (service *HTTPRestService) getNetworkContainerResponse(req cns.GetNetworkCo
 		}
 
 		logger.Printf("pod info %+v", podInfo)
-		containerID = service.state.ContainerIDByOrchestratorContext[podInfo.PodName+podInfo.PodNamespace]
+
+		var (
+			context      = podInfo.PodName + podInfo.PodNamespace
+			dncEP        = service.GetOption(acn.OptPrivateEndpoint).(string)
+			infraVnet    = service.GetOption(acn.OptInfrastructureNetwork).(string)
+			nodeID       = service.GetOption(acn.OptNodeID).(string)
+			isManagedDnc = dncEP != "" && infraVnet != "" && nodeID != ""
+		)
+
+		containerID, exists = service.state.ContainerIDByOrchestratorContext[context]
+		if !exists {
+			if isManagedDnc {
+				service.lock.Unlock()
+				getNetworkContainerResponse.Response.ReturnCode, getNetworkContainerResponse.Response.Message = service.SyncNodeStatus(dncEP, infraVnet, nodeID, req.OrchestratorContext)
+				service.lock.Lock()
+				if getNetworkContainerResponse.Response.ReturnCode == NotFound {
+					return getNetworkContainerResponse
+				}
+
+				containerID = service.state.ContainerIDByOrchestratorContext[context]
+			}
+		} else if isManagedDnc {
+			_, getNetworkContainerResponse.Response.ReturnCode, getNetworkContainerResponse.Response.Message = service.isNCWaitingForUpdate(service.state.ContainerStatus[containerID].CreateNetworkContainerRequest.Version, containerID)
+		}
```

Comment:
move to separate function

Reply:
move what to separate function?

Reply:
github doesnt have good gui...

`Comment on lines 1266 to 1289`